documentation/External-Grafana/ReadMe.md

```diff
@@ -0,0 +1,179 @@
+This includes design elements to support the following with Grafana operator :
+- Ability to define an external Grafana instance as Grafana source to Grafana operator
+- Ability to create Grafana dashboards on a remote Grafana Instance.
+- Ability to add external cloud data sources to an external Grafana instance.
+- Ability to create alerting on a remote Grafana Instance.
+
+## Info
+
+status: Draft
+
+## Motivation
+
+Cloud providers such Amazon Web Services (AWS) have started providing managed remote grafana services such as Amazon Managed Grafana which decouples the responsibilities of managing a grafana instanaces from ops personas and kubernetes environment. 
+
+Currently Grafana operator has an integration to add Amazon Managed Service for Promethus (AMP) as a data source to Grafana instances hosted in a kubernetes environment. As more customers starting to use external grafana services, expanding the Grafana operator to support remote Grafana instances becomes inevitable. Adding ability to integrate with external grafana services, adding data sources, creating dashboards and alerting on a remote Grafana instances offloads responsibilities of managing a grafana instanaces from ops personas which helps them to focus on developing the features required for their business. This helps the customer teams to move from self managed Grafana instance on their Kubernetes environments to Pay as you go model on Grafana instances provided by providers.
+
+## Out of scope
+
+In the design, integration to remote managed Grafana providers other than Amazon Managed Grafana is out of scope at this point. We may add additional integrations as part of future road map items.
+
+## Verification
+
+- Create integration tests for adding keys for remote Grafana instance.
+- Create integration tests to create Grafana dashboards on a remote Grafana Instance.
+- Create integration tests to add cloud data sources such as Amazon Managed Service for Prometheus, Amazon CloudWatch to remote Grafana instance.
+- Create integration tests to create alerting on a remote Grafana Instance.
+
+## Current
+
+Currently the grafana operator supports the following for only self managed Grafana Instance :
+- Adding remote data sources such as Amazon Managed Service for Prometheus.
+- Creating Dashboards.
+- Setting up alerting.
+- And Many more.
+
+## Proposal
+
+In short the proposal in this document is about enhancing the Grafana Operator to support the integration to managed grafana services. We would need to enhance the current version of Grafana Operator to support the following :
+
+- Defining external Grafana instance as Grafana source to Grafana operator.
+- Creating Grafana dashboards on external Grafana instance.
+- Adding data sources to external Grafana instance.
+- Creating alerting on external Grafana instance.
+
+### Defining external Grafana instance as Grafana source to Grafana operator.
```

Comment:
We don't think we need a new CRD, we should be able to use the existing one.
I have provided an example below.

Reply:
I have made this change.